test/test_encoding_utils.rb

```diff
@@ -52,7 +52,9 @@ def test_fail_to_unmarshal_older_major_version
   end
 
   def test_fail_to_unmarshal_not_enough_data
-    skip "TypeError on TruffleRuby" if RUBY_ENGINE == "truffleruby"
+    if RUBY_ENGINE == 'truffleruby' and Gem::Version.new(RUBY_ENGINE_VERSION) < Gem::Version.new('23.0.0.a')
```

Comment:
Alternative: if we don't want these checks and just want to remove the skips, then we test only on truffleruby-dev, and not on `truffleruby` (release), until the next release.

Reply:
How far away until the next release? As in it is worth waiting for that next release and remove all the skips?

Reply:
Only in April, in ~2 months: https://www.graalvm.org/release-notes/release-calendar/
I'd much rather it's merged before that if possible.

The current PR is ideal because it doesn't need any change once 23.0 is released.

Reply:
Fine, by me, but I'd appreciate if you could open another PR later to remove these.

Reply:
OK, I created a reminder to do that after the 23.0 release.